graph/patterns/longRunningOperations.md

```diff
@@ -0,0 +1,192 @@
+# Long running operations
+
+Microsoft Graph API Design Pattern
+
+__*The long running operations (LRO) pattern provides the ability to model operations where processing a client request takes a long time, but the client isn't blocked and can do some other work until operation completion.*__
+
+## Problem
+
+The API design requires modeling operations on resources, which takes a long time
+to complete so that API clients don't need to wait and can continue doing other
+work while waiting for the final operation results. The client should be able to
+monitor the progress of the operation and have an ability to cancel it if
+needed. The API needs to provide a mechanism to track the work
+being done in the background. The mechanism needs to be expressed in the same
+web style as other interactive APIs. It also needs to support checking on the status and/or
+being notified asynchronously of the results.
+
+## Solution
+
+The solution is to model the API as a synchronous service that returns a
+resource that represents the eventual completion or failure of a long running
+operation.
+
+There are two flavors of this solution:
+
+- The returned resource is the targeted resource and includes the status of
+    the operation. This pattern is often called RELO (resource-based
+    long running operation).
+
+<!-- markdownlint-disable MD033 -->
+<p align="center">
+  <img src="RELO.gif" alt="The status monitor LRO flow"/>
+</p>
+<!-- markdownlint-enable MD033 -->
+
+- The returned resource is a new API resource called 'Stepwise Operation' and
+    is created to track the status. This LRO solution is similar to the concept
+    of Promises or Futures in other programming languages.
+
+<!-- markdownlint-disable MD033 -->
+<p align="center">
+  <img src="LRO.gif" alt="The status monitor LRO flow"/>
+</p>
+<!-- markdownlint-enable MD033 -->
+
+The RELO pattern is the preferred pattern for long running operations and should be
+used wherever possible. The pattern avoids complexity, and consistent resource
+presentation makes things simpler for our users and tooling chain.
+
+In general, Microsoft Graph API guidelines for LRO follow [Microsoft REST API
+Guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#13-long-running-operations).
+
+A single deviation from the base guidelines is that Microsoft Graph API standards require that you use the following response headers:
```

Comment:
Do we also want to call out that we are not allowing API producers to expose a tenant-level /operations resource (as the main API guidelines allow in https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#1324-operations-resource)

Reply:
Thank you! good point. will update the pattern